src/finch/compiled.py

```diff
@@ -0,0 +1,34 @@
+from functools import wraps
+
+from .julia import jl
+from .tensor import Tensor
+
+
+def compiled(func):
+    @wraps(func)
+    def wrapper_func(*args, **kwargs):
+        new_args = []
+        for arg in args:
+            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
+                new_args.append(Tensor(jl.Finch.LazyTensor(arg._obj)))
+            else:
+                new_args.append(arg)
+
+        result = func(*new_args, **kwargs)
+        result_tensor = Tensor(jl.Finch.compute(result._obj))
+
+        return result_tensor
+
+    return wrapper_func
+
+
+def lazy(tensor: Tensor):
+    if not jl.isa(tensor._obj, jl.Finch.LazyTensor):
```

Comment:
I'd suggest adding an `is_lazy()` method to `Tensor` to compute exactly this, and using it here for a cleaner interface. You can also do a function if you prefer.

Reply:
@hameerabbasi Did you mean `to_lazy()`? I think `is_lazy` refers to a bool check.

Reply:
I indeed mean a method to check if a given `Tensor` is lazy/computed. We could add an `is_computed()` method instead as well, which I think would be a better idea.

Reply:
Done!

Reply:
I think this is an instance where the underlying library should also handle this logic. Because e.g. jl.lazy(LazyTensor) is already a no-op and jl.compute(swizzle) should be a no-op if it's not already, can we opt to simplify this logic in a later PR so that we simply always call jl.compute or always call jl.lazy?

Reply:
Because Finch didn't have a lazy interface when we started, I think this project is an opportunity to explore what that interface should look like in both of the libraries, so if it makes sense to add functionality on the python side, it probably makes sense to add it on the julia side too.

Reply:
That's a good point - `jl.lazy(anything)` should decide whether wrapping happens or not. I can make these adjustment in the next PR.

Reply:
I'm also gonna make a PR to Finch.jl to make sure that if any non-lazy tensors are passed to compute they get returned as-is